### ISSUE TITLE
Node module warnings

### ISSUE THREAD

**Issue by pbower** [ghiscoding/slickgrid-react#273]
### Describe the bug

Hi there,

Merry Christmas - trust you're looking forward to the holiday season.

Sharing back node warnings, that look like they relate to JS files referencing non-existent src locations.

It's possible this is a setup thing, but I've grabbed the latest versions from npm and experiencing the (389) warnings, so I thought I'd feed it back.

It occurs with npm run react-star, and I am using one non-interesting SlickgridReact instance.

If any further details are required please let me know.

Thanks!

<img width="213" alt="image" src="https://github.com/ghiscoding/slickgrid-react/assets/37089506/971981a4-1139-4ead-8723-f78e2bf87d93">

WARNING in ./node_modules/slickgrid-react/dist/esm/services/translater.service.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '.../node_modules/slickgrid-react/src/slickgrid-react/services/translater.service.ts' file: Error: ENOENT: no such file or directory, open '.../client/node_modules/slickgrid-react/src/slickgrid-react/services/translater.service.ts'

WARNING in ./node_modules/slickgrid-react/dist/esm/services/utilities.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '.../node_modules/slickgrid-react/src/slickgrid-react/services/utilities.ts' file: Error: ENOENT: no such file or directory, open '.../node_modules/slickgrid-react/src/slickgrid-react/services/utilities.ts'

WARNING in ./node_modules/slickgrid-react/dist/esm/slickgrid-config.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '.../node_modules/slickgrid-react/src/slickgrid-react/slickgrid-config.ts' file: Error: ENOENT: no such file or directory, open '.../node_modules/slickgrid-react/src/slickgrid-react/slickgrid-config.ts'

### Reproduction

slickgrid-react@4.0.2
── @slickgrid-universal/common@4.0.3
├── @slickgrid-universal/composite-editor-component@4.0.3
├── @slickgrid-universal/custom-tooltip-plugin@4.0.3
├── @slickgrid-universal/empty-warning-component@4.0.3
├── @slickgrid-universal/event-pub-sub@4.0.2
├── @slickgrid-universal/excel-export@4.0.3
├── @slickgrid-universal/row-detail-view-plugin@4.0.3
├── @slickgrid-universal/utils@4.0.2

### Expectation

No warnings.

### Environment Info

```shell
Typescript
VsCode
├── react@18.2.0

  System:
    OS: Linux 5.15 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (24) x64 12th Gen Intel(R) Core(TM) i9-12900KF
    Memory: 26.23 GB / 31.24 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.5.1 - /usr/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 10.2.0 - /usr/bin/npm
  npmPackages:
    slickgrid-react: ^4.0.2 => 4.0.2
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/slickgrid-react/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [HOWTO - Step by Step](https://github.com/ghiscoding/slickgrid-react/wiki/HOWTO--Step-by-Step).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/slickgrid-react/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/slickgrid-react/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
It really looks like it wants me to include the source files and this is the second time you talk about source files, it might be needed by the source map maybe. I review the .npmignore again and reinclude the source file if it's really a problem. I've never seen these warnings on my side though and I assume you only get them while running Dev not Prod

**Comment by pbower** [comment]
Thanks for your response.
Ok that's interesting.
Re. the source files, that was likely more my inexperience than a standard pattern. In this case, are there some loose references in the JS?

**Comment by ghiscoding** [comment]
Well I mean it's showing in your warnings that you posted and then I looked at my project source maps in the `dist` folder and indeed it is pointing to `src` and I can troubleshoot locally and add breakpoints but the fact is that I always excluded the `src` folder from npm publish so...

here's 1 of your warning, I highlighted the `src` (though it might be hard to see below)

> WARNING in ./node_modules/slickgrid-react/dist/esm/services/translater.service.js
> Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
> Failed to parse source map from '.../node_modules/slickgrid-react/**src**/slickgrid-react/services/translater.service.ts' file: Error: ENOENT: no such file or directory, open '.../client/node_modules/slickgrid-react/**src**/slickgrid-react/services/translater.service.ts'

then if you use the `unpkg` website which is very helpful to see what is published, you can see exactly which files and folders are part of the npm package and `src` was never included, e.g.
https://unpkg.com/browse/@slickgrid-universal/common@4.0.3/

I already made a PR for Slickgrid-Universal to add the `src` folder, I have to do the same in this repo here as well

**Comment by pbower** [comment]
Thanks, ok sounds great.

**Comment by ghiscoding** [comment]
The PRs in both repos should fix the issue but I won't publish until a couple more days

**Comment by ghiscoding** [comment]
This should be fixed now with latest v4.1.0, the `src` is now included for Slickgrid-React and all Slickgrid-Universal packages

**Comment by pbower** [comment]
Awesome.
